packages/jellyfish-api-core/__tests__/category/icxorderbook/makeOffer.test.ts

```diff
@@ -0,0 +1,467 @@
+    expect(Object.keys(orders).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    expect((orders as Record<string, ICXOfferInfo>)[makeOfferTxId]).toStrictEqual(
+      {
+        orderTx: createOrderTxId,
+        status: ICXOrderStatus.OPEN,
+        amount: offer.amount,
+        amountInFromAsset: offer.amount.dividedBy(order.orderPrice),
+        ownerAddress: offer.ownerAddress,
+        takerFee: offer.amount.multipliedBy(ICX_TAKERFEE_PER_BTC),
+        expireHeight: expect.any(BigNumber),
+        receivePubkey: offer.receivePubkey
+      }
+    )
+
+    // check accountDFI[idDFI] balance
+    const accountDFIAfterOffer = await container.call('getaccount', [accountDFI, {}, true])
+    expect(accountDFIAfterOffer).toStrictEqual(accountDFIBeforeOrder)
+  })
+
+  it('make a higher offer to an sell DFI order, should occur equivalent taker fee', async () => {
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = result.txid
+    await container.generate(1)
+
+    // list ICX orders
+    let orders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
+    expect((orders as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
+      {
+        status: ICXOrderStatus.OPEN,
+        type: ICXOrderType.INTERNAL,
+        tokenFrom: symbolDFI,
+        chainTo: order.chainTo,
+        receivePubkey: order.receivePubkey,
+        ownerAddress: order.ownerAddress,
+        amountFrom: order.amountFrom,
+        amountToFill: order.amountFrom,
+        orderPrice: order.orderPrice,
+        amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
+        height: expect.any(BigNumber),
+        expireHeight: expect.any(BigNumber)
+      }
+    )
+
+    const accountBTCBeforeOffer = await container.call('getaccount', [accountBTC, {}, true])
+    // make offer to higher amout 20 DFI - taker
+    const offer: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.20), // 0.20 BTC = 20 DFI
+      ownerAddress: accountBTC
+    }
+    result = await client.icxorderbook.makeOffer(offer, [])
+    const makeOfferTxId = result.txid
+    await container.generate(1)
+
+    const accountBTCAfterOffer = await container.call('getaccount', [accountBTC, {}, true])
+    // check fee of 0.02 DFI has been reduced from the accountBTCBeforeOffer[idDFI]
+    // Fee = takerFeePerBTC(inBTC) * amount(inBTC) * DEX DFI per BTC rate
+    expect(accountBTCAfterOffer[idDFI].toPrecision(8)).toStrictEqual((accountBTCBeforeOffer[idDFI] - 0.02).toPrecision(8))
```

Comment:
@canonbrother I remember you asked about this in a comment. Not sure what happened to that comment. But bringing it upfront again. Just check out this branch and remove both `toPrecision(8)`s and then run the test.
I'm getting this once it's removed.
```
FAIL  packages/jellyfish-api-core/__tests__/category/icxorderbook/makeOffer.test.ts (31.442 s)
  ● ICXOrderBook.makeOffer › make a higher offer to an sell DFI order, should occur equivalent taker fee

    expect(received).toStrictEqual(expected) // deep equality

    Expected: 2009.8700000000001
    Received: 2009.87

      219 |     // Fee = takerFeePerBTC(inBTC) * amount(inBTC) * DEX DFI per BTC rate
      220 |     // expect(accountBTCAfterOffer[idDFI].toPrecision(8)).toStrictEqual((accountBTCBeforeOffer[idDFI] - 0.02).toPrecision(8))
    > 221 |     expect(accountBTCAfterOffer[idDFI]).toStrictEqual(accountBTCBeforeOffer[idDFI] - 0.02)
          |                                         ^
      222 |
      223 |     // List the ICX offers for orderTx = createOrderTxId and check
      224 |     orders = await client.call('icx_listorders', [{ orderTx: createOrderTxId }], 'bignumber')

```

Reply:
Expected `2009.8700000000001` is crazy
its 13 decimal places which should be out of support from ain (max 8 for now)

Reply:
we don't need `toPrecision(8)` which just make the figure from
`2009.87` to just `2009.8700`
use `toFixed()` is more suitable

Reply:
yes looks like a problem in internal representation after an arithmetic operation. I guess no option other than comparing after `toPrecision(8)`

Reply:
see https://github.com/DeFiCh/jellyfish/pull/399/files#r660450894

you are losing precision because you use `number`

Reply:
using `bignumber` resolved this problem also